core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactory.java

```diff
@@ -92,6 +92,14 @@ RelDataType createStructType(
       List<RelDataType> typeList,
       List<String> fieldNameList);
 
+  /** Creates a user defined type that represents a structured collection of fields with a name.
+   *
+   **/
+  RelDataType createStructuredType(
```

Comment:
rename  "createStructuredTypeWithName" and "createArrayTypeWithName"

Reply:
renamed